### PR TITLE
Soft-Delete

### PR DESCRIPTION
**Pull request by ntjhal**
Gave all database tracked entities a status of 'Active' or 'Archived'. I've chosen to work with an enum in code since its likely to be a one time thing in this operation and doesn't have to be tracked in the db because of that. 

After a delete the items state goes to archived. This means you will not get the item when retrieving a list but you can still request it on its unique ID. 

Hope this is what we are looking for

image
image

### CODE REVIEW

**Review by 0xboomie — APPROVED** [review]
Goed werk!

File: dataservice/Application/Repositories/ColorRepo.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public Result AddColor(string color)
 
         public IEnumerable<MedicineColor> GetColors()
         {
-            return _context.Medicine_Colors;
+            return _context.Medicine_Colors.Where(x => x.Status == Domain.EntityStatus.Active.ToString().ToLower().ToString().ToLower());
```
Comment (MerijnMonfils) [inline dataservice/Application/Repositories/ColorRepo.cs:42]: wth is dat laatste? ToString -> ToLower en dan nog een keer?

Comment (MerijnMonfils) [reply dataservice/Application/Repositories/ColorRepo.cs:42]: vanwaar het dubbele? Ik zie het namelijk op meerdere plekken

**Review by MerijnMonfils — APPROVED** [review]
paar comments op de code verder netjes :)

File: dataservice/Application/Repositories/ColorRepo.cs (hunk 1 above)
Comment (ntjhal) [reply dataservice/Application/Repositories/ColorRepo.cs:42]: oeps, ctrl + f replace, niet goed gecontroleerd van mezelf

**Pull request merged**